Features/paginering.feature

```diff
@@ -8,96 +8,137 @@ Functionaliteit: Als afnemer wil ik kunnen bladeren door een groot aantal result
+    En bevat het antwoord geen attribuut _links.previous
+    En bevat het antwoord geen attribuut _links.next
+    En bevat het antwoord geen attribuut _links.last
 
   Scenario: de zoekvraag levert één pagina
-    Als de request wordt gedaan naar een endpoint waar paginering van toepassing is
-    En er zitten er voldoen minder dan 20 items in het antwoord
-    Dan is attribuut _links.self.href gelijk aan de request
+    Als de zoekvraag wordt gedaan op een endpoint waar paginering van toepassing is
+    En er zijn 1-20 resultaten die voldoen aan de zoekvraag
+    Dan bevat het antwoord alle gevonden resultaten in de collectie
+    En is attribuut _links.self.href opgebouwd zoals beschreven in de links.feature: https://github.com/lvbag/BAG-API/blob/master/Features/links.feature
     En bevat het antwoord geen attribuut _links.first
     En bevat het antwoord geen attribuut _links.previous
     En bevat het antwoord geen attribuut _links.next
     En bevat het antwoord geen attribuut _links.last
 
   Scenario: de zoekvraag levert geen resultaten
-    Als de request wordt gedaan naar een endpoint waarbij paginering van toepassing is
+    Als de zoekvraag wordt gedaan op een endpoint waarbij paginering van toepassing is
     En er zijn geen resulaten gevonden
-    Dan is attribuut _links.self.href gelijk aan de request
+    Dan bevat het antwoord geen collectie
+    En is attribuut _links.self.href gelijk aan de request
     En bevat het antwoord geen attribuut _links.first
     En bevat het antwoord geen attribuut _links.previous
     En bevat het antwoord geen attribuut _links.next
     En bevat het antwoord geen attribuut _links.last
 
-  Scenario: de zoekvraag levert meerdere pagina's en de pageSize parameter wordt gebruikt
-    Als de request wordt gedaan met parameter "?page=3&pageSize=15"
-    Dan zitten er 15 resultaten in het antwoord
-    En is attribuut _links.self.href gelijk aan "?page=3&pageSize=15"
-    En is attribuut _links.first.href gelijk aan "?page=1&pageSize=15"
-    En is attribuut _links.previous.href gelijk aan "?page=2&pageSize=15"
-    En is attribuut _links.next gelijk aan "?page=4&pageSize=15"
-    En is attribuut _links.last gelijk aan "?page=5&pageSize=15"
-    
-   
+  Scenario: de zoekvraag levert meerdere pagina's en de page en pageSize parameters worden gebruikt
+    Als de zoekvraag wordt gedaan op een endpoint waarbij paginering van toepassing is
+    En de zoekvraag bevat een parameter "page=3"
+    En de zoekvraag bevat een parameter "pageSize=15"
+    En er zijn 61-75 resultaten die voldoen aan de zoekvraag
+    Dan bevat het antwoorden resultaten 31 t/m 45 in de collectie
+    En is attribuut _links.self.href opgebouwd zoals beschreven in de links.feature: https://github.com/lvbag/BAG-API/blob/master/Features/links.feature "?page=3&pageSize=15"
+    En bevat het antwoord een attribuut _links.first.href met "<de bron URI zonder de opgegeven page parameter>"
```

Comment:
link bevat ook PageSize parameter

Reply:
Dat klopt, maar de self link wordt opgebouwd zoals beschreven in de links.feature, dat betekent, dat de opgegeven page en pageSize parameter er in zitten. Echter is het zo dat de page parameter wordt vervangen door een andere waarde, vandaar dat hier wordt aangegeven dat de self link <de bron URI zonder de opgegeven page parameter> is (daar zit de pageSize dus nog wel in en vervolgens wordt er aangegeven, wat de page parameter wordt.